sensor/common/scan/scan.go

```diff
@@ -270,10 +348,17 @@ func (s *LocalScan) fetchSignatures(ctx context.Context, errorList *errorhelpers
 	// Fetch signatures from cluster-local registry.
 	sigs, err := s.fetchSignaturesWithRetry(ctx, signatures.NewSignatureFetcher(), image, image.GetName().GetFullName(), registry)
 	if err != nil {
-		log.Debugf("Failed fetching signatures for image %q: %v", image.GetName().GetFullName(), err)
+		// Like Central, only log errors related to fetching signatures.
+		if !errors.Is(err, errox.NotAuthorized) {
+			log.Errorf("Fetching image signatures for image %q: %v", image.GetName().GetFullName(), err)
+		} else {
+			// Log not authorized errors in debug mode, since we expect them to occur.
+			log.Debugf("Unauthorized error fetching image signatures for image %q: %v", image.GetName().GetFullName(), err)
+		}
+	}
+
+	if len(sigs) == 0 {
 		image.Notes = append(image.Notes, storage.Image_MISSING_SIGNATURE)
-		errorList.AddError(pkgErrors.Wrapf(err, "fetching signature for image %q from registry %q", image.GetName(), registry.Name()))
```

Comment:
why do we even bother calling this function if we have errors? Especially now that we don't use the errorList, I think we `should` remove the need for it

Reply:
I believe it keeps the upper function more readable, for example, currently:

```go
	// Perform partial scan (image analysis / identify components) via local scanner.
	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, pullSourceImage)

	// Fetch signatures associated with image from registry.
	sigs := s.fetchSignatures(ctx, errorList, reg, pullSourceImage)

	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
	// signature verification results.
	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
```

With adding an error check:

```go
	// Perform partial scan (image analysis / identify components) via local scanner.
	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, pullSourceImage)

	// Fetch signatures associated with image from registry.
	if errorList.Empty() {
		sigs := s.fetchSignatures(ctx, errorList, reg, pullSourceImage)
	}
	
	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
	// signature verification results.
	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
```

I believe I've seen it called `line of sight`, the idea is it helps readability to keep the happy path indented at the same level, and the error path at a separate indent.